manim/mobject/geometry.py

```diff
@@ -1428,143 +1432,313 @@ def __init__(self, start_anchor, start_handle, end_handle, end_anchor, **kwargs)
-        """Gets the vertices of the polygon.
+    def get_vertices(self) -> np.ndarray:
+        """Gets the vertices of the :class:`Polygram`.
+
+        Returns
+        -------
+        :class:`numpy.ndarray`
+            The vertices of the :class:`Polygram`.
 
         Examples
         --------
         ::
 
             >>> sq = Square()
-            >>> points = sq.get_vertices()
-            >>> points
+            >>> sq.get_vertices()
             array([[ 1.,  1.,  0.],
                    [-1.,  1.,  0.],
                    [-1., -1.,  0.],
                    [ 1., -1.,  0.]])
+        """
+
+        return self.get_start_anchors()
+
+    def get_vertex_groups(self) -> np.ndarray:
+        """Gets the vertex groups of the :class:`Polygram`.
 
         Returns
         -------
         :class:`numpy.ndarray`
-            Returns a list of the coordinates of polygon's vertices.
+            The vertex groups of the :class:`Polygram`.
+
+        Examples
+        --------
+        ::
+
+            >>> poly = Polygram([ORIGIN, RIGHT, UP], [LEFT, LEFT + UP, 2 * LEFT])
+            >>> poly.get_vertex_groups()
+            array([[[ 0.,  0.,  0.],
+                    [ 1.,  0.,  0.],
+                    [ 0.,  1.,  0.]],
+            <BLANKLINE>
+                   [[-1.,  0.,  0.],
+                    [-1.,  1.,  0.],
+                    [-2.,  0.,  0.]]])
         """
 
-        return self.get_start_anchors()
+        vertex_groups = []
+
+        group = []
+        for start, end in zip(self.get_start_anchors(), self.get_end_anchors()):
+            group.append(start)
+
+            if self.consider_points_equals(end, group[0]):
+                vertex_groups.append(group)
+                group = []
+
+        return np.array(vertex_groups)
 
-    def round_corners(self, radius=0.5):
-        """Rounds off the corners of the polygon.
+    def round_corners(self, radius: float = 0.5):
+        """Rounds off the corners of the :class:`Polygram`.
 
         Parameters
         ----------
-        radius : :class:`float`, optional
-            The curvature of the corners of the polygon.
+        radius
+            The curvature of the corners of the :class:`Polygram`.
 
         Examples
         --------
 
-        .. manim:: PolygonRoundCorners
-            :save_as_gif:
+        .. manim:: PolygramRoundCorners
+            :save_last_frame:
 
-            class PolygonRoundCorners(Scene):
+            class PolygramRoundCorners(Scene):
                 def construct(self):
-                    points = [[-4, -2, 0], [-2, 2, 0], [4, 2, 0], [2, -2, 0]]
-                    parallelogram = Polygon(*points, stroke_color=LIGHT_PINK)
-                    rounded_1 = Polygon(*points, stroke_color=LIGHT_PINK).round_corners(radius=0.5)
-                    rounded_2 = Polygon(*points, stroke_color=LIGHT_PINK).round_corners(radius=1.5)
+                    square = Polygram([[1, 1, 0], [-1, 1, 0], [-1, -1, 0], [1, -1, 0]])
```

Comment:
I would like it if the Polygram used head corners with different angles.

Reply:
I'm unsure of what you mean with "head corners"

Reply:
*had :laughing: